walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/requirements/constraints/ClaimRegexConstraint.kt

```diff
@@ -0,0 +1,46 @@
+package com.microsoft.walletlibrary.requests.requirements.constraints
+
+import com.jayway.jsonpath.Configuration
+import com.jayway.jsonpath.JsonPath
+import com.jayway.jsonpath.Option
+import com.microsoft.did.sdk.credential.models.VerifiableCredentialContent
+import com.microsoft.did.sdk.util.log.SdkLog
+import com.microsoft.walletlibrary.util.VerifiedIdIssuerIsNotRequestedException
+import com.microsoft.walletlibrary.verifiedid.VerifiableCredential
+import com.microsoft.walletlibrary.verifiedid.VerifiedId
+import kotlinx.serialization.json.Json
+import java.util.regex.Pattern
+
+class ClaimRegexConstraint(internal val path: List<String>, internal val pattern: String): VerifiedIdConstraint {
+    override fun doesMatch(verifiedId: VerifiedId): Boolean {
+        if (verifiedId !is VerifiableCredential)
+            return false
+        val vccJsonString = Json.encodeToString(VerifiableCredentialContent.serializer(), verifiedId.raw.contents)
+        return path.any { matchAnyPathInFields(pattern, it, vccJsonString) }
+    }
+
+    private fun matchAnyPathInFields(pattern: String, path: String, vccJson: String): Boolean {
+        if (pattern.isEmpty())
+            SdkLog.d("Empty pattern in filter.")
+        val configuration = Configuration.defaultConfiguration().addOptions(Option.SUPPRESS_EXCEPTIONS)
+        val constraintValue: String? = JsonPath.using(configuration).parse(vccJson).read(path)
+        return if (constraintValue != null) {
+            matchPattern(pattern, constraintValue)
+        } else
+            false
+    }
+
+    internal fun matchPattern(pattern: String, value: String): Boolean {
+        val sanitizedPattern = sanitizePattern(pattern)
+        return Pattern.compile(sanitizedPattern, Pattern.CASE_INSENSITIVE).matcher(value).find()
+    }
+
+    private fun sanitizePattern(pattern: String): String {
+        return pattern.split("/").firstOrNull { it.isNotEmpty() } ?: ""
```

Comment:
I'm not sure that this is always safe unless you're trying to avoid allowing a path-like value? If I understand the intention, you want the pattern (`regex`) in a string like `/regex/gi` but not the flags? In which case this would break a regex like `foo\/bar` to match "foo/bar"?

Reply:
That was a good catch Logan. I have fixed it and added another test for it as well :)